src/procedures/__tests__/AssignSecurityTokenRole.ts

```diff
@@ -0,0 +1,267 @@
+      'SecurityTokenReservationFactory'
+    );
+    shareholderFactoryMock = ImportMock.mockClass(shareholderFactoryModule, 'ShareholderFactory');
+    usdTieredStoFactoryMock = ImportMock.mockClass(
+      usdTieredStoFactoryModule,
+      'UsdTieredStoFactory'
+    );
+    taxWithholdingFactoryMock = ImportMock.mockClass(
+      taxWithholdingFactoryModule,
+      'TaxWithholdingFactory'
+    );
+
+    securityTokenFactoryMock.mock('fetch', {
+      permissions: {
+        isRoleAvailable: () => Promise.resolve(true),
+        getFeatureFromRole: () => Promise.resolve(true),
+      },
+    });
+
+    gpmMock.mock('getAllDelegates', Promise.resolve([params1.delegateAddress]));
+    gpmMock.mock('getAllDelegatesWithPerm', Promise.resolve([]));
+
+    getAttachedModulesMockStub = wrappersMock.mock(
+      'getAttachedModules',
+      Promise.resolve([gpmMock.getMockInstance()])
+    );
+    tokenFactoryMockStub = tokenFactoryMock.mock('getSecurityTokenInstanceFromTicker', {});
+    const factoryMockSetup = {
+      securityTokenFactory: securityTokenFactoryMock.getMockInstance(),
+      securityTokenReservationFactory: securityTokenReservationFactoryMock.getMockInstance(),
+      erc20TokenBalanceFactory: erc20TokenBalanceFactoryMock.getMockInstance(),
+      investmentFactory: investmentFactoryMock.getMockInstance(),
+      cappedStoFactory: cappedStoFactoryMock.getMockInstance(),
+      usdTieredStoFactory: usdTieredStoFactoryMock.getMockInstance(),
+      dividendDistributionFactory: dividendDistributionFactoryMock.getMockInstance(),
+      checkpointFactory: checkpointFactoryMock.getMockInstance(),
+      erc20DividendsManagerFactory: erc20DividendsManagerFactoryMock.getMockInstance(),
+      ethDividendsManagerFactory: ethDividendsManagerFactoryMock.getMockInstance(),
+      shareholderFactory: shareholderFactoryMock.getMockInstance(),
+      taxWithholdingFactory: taxWithholdingFactoryMock.getMockInstance(),
+    };
+    contextMock.set('factories', factoryMockSetup);
+
+    wrappersMock.mock('roleToPermission', {
+      moduleName: ModuleName.PercentageTransferManager,
+      permission: Perm.Operator,
+    });
+    wrappersMock.mock('getModuleAddressesByName', [params1.delegateAddress]);
+
+    // Instantiate AssignSecurityTokenRole
+    target = new AssignSecurityTokenRole(
+      {
+        symbol: params1.symbol,
+        delegateAddress: params1.delegateAddress,
+        role: params1.role,
+        assign: params1.assign,
+      },
+      contextMock.getMockInstance()
+    );
+  });
+  afterEach(() => {
+    sinon.restore();
+  });
+  describe('Types', () => {
+    test('should extend procedure and have AssignSecurityTokenRole type', async () => {
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe('AssignSecurityTokenRole');
+    });
+  });
+
+  describe('AssignSecurityTokenRole', () => {
+    test('should send the transaction to AssignSecurityTokenRole', async () => {
```

Comment:
This description should say something like

"should enqueue the addDelegate and changePermission transactions when the delegate is a new address"